src/main/java/org/cryptomator/cryptofs/health/shortened/ShortenedNamesCheck.java

```diff
@@ -103,6 +115,36 @@ void checkShortenedName(Path dir) throws IOException {
 			}
 		}
 
+
+		/**
+		 * Checks if the string stored inside the name file is base64url encoded
+		 *
+		 * <em>visible for testing</em>
+		 *
+		 * @return {@link NameEncoding} indicating if it is valid, invalid or affected by https://github.com/cryptomator/cryptofs/issues/121
+		 */
+		NameEncoding isValidEncodedName(String toAnalyse) {
+			int firstNullByte = toAnalyse.indexOf('\0');
+			var beforeFirstNull = firstNullByte == -1 ? toAnalyse : toAnalyse.substring(0, firstNullByte);
+			var afterFirstNull = firstNullByte == -1 ? "" : toAnalyse.substring(firstNullByte);
+
+			boolean isFirstBase64 = BASE64URL.canDecode(beforeFirstNull);
+
+			if (isFirstBase64 && afterFirstNull.isEmpty()) {
+				return NameEncoding.VALID;
+			} else if (isFirstBase64 && NULL_BYTES.matcher(afterFirstNull).matches()) {
+				return NameEncoding.BUG121;
+			} else {
+				return NameEncoding.INVALID;
+			}
+		}
+
+		enum NameEncoding {
+			VALID,
+			INVALID,
+			BUG121;
```

Comment:
I'd prefer a name that doesn't require you to know the issue description. A link to the github issue in a comment would be nice, though.

Suggestion:

```suggestion
			NULL_PADDED; // contains trailing 0x00 bytes, see https://github.com/cryptomator/cryptofs/issues/121
```

Reply:
Considered for db83548 , I renamed it to `TRAILING_BYTES`